### DISCUSSION TITLE
Getting error when creating containers

### DISCUSSION THREAD

**Discussion by AlphaNecron** [criyle/go-judge#85] [Q&A] [answered]
I'm getting this when running the program on Debian, any ideas? Running as root btw. With `linux` 6.5.
`conf: recvAck RecvMsg: read unixpacket @->@: EOF`

**Comment by AlphaNecron** [comment]
![image](https://github.com/criyle/go-judge/assets/57827456/fbd7ef56-ba9f-4761-b986-ff55dde86354)
When enabling stderr, the error appears. Works fine on Arch but not Debian.

**Comment by AlphaNecron** [comment]
Nevermind, fixed, `root` of the container was not found. 
![image](https://github.com/criyle/go-judge/assets/57827456/f1d839e4-bf49-419a-9e87-e5484078b574)
However, there's another issue. I can't bind /etc/passwd, any solutions?

**Reply by criyle** [reply]
Maybe the `passwd` file is not accessible to everybody or the filesystem is not bindable. You may skip the `passwd` file as it only provides a user name when executing command like `ls -l` or `who am I`.

**Comment by AlphaNecron** [comment]
Is `/etc/passwd` mandatory? My program seems to work fine without that.

**Reply by criyle** [reply]
No, it is optional to the container. You may control which file to be bind mounted into container by `mount.yaml` file.

**Reply by AlphaNecron** [reply]
Thanks. Is it possible to dynamically mount a specific file at runtime?

**Reply by criyle** [reply]
No, container are being reused and mount namespace was frozen after created.

**Reply by AlphaNecron** [reply]
Oh, how about symlinks? Can we do symlinks from host to containers at runtime and unlink them later?

**Reply by criyle** [reply, marked as answer]
No, the container is running with a readonly system root with `/w` and `/tmp` mounted with `tmpfs`, which are being cleared after every execution. At the same time, symbolic from host to container is considered as a dangerous operation as they may able to alter its content or escape from the isolated mount namespace. You may use the `copyIn` argument to copy file content into the `tmpfs` within the container and it is rather a lightweight operation thanks to the file page cache.

**Reply by AlphaNecron** [reply]
Thanks, I'm gonna try passing the file to /dev/stdin instead.
